### PR TITLE
Add a singleton, frozen empty array to use in place of [] which create/g...

### PR DESCRIPTION
**Pull request by marchant**
...arbage one everytime

### CODE REVIEW

File: generic-collection.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ function GenericCollection() {
     throw new Error("Can't construct. GenericCollection is a mixin.");
 }
 
+GenericCollection.EmptyArray = Object.freeze([]);
```
Comment (kriskowal) [inline generic-collection.js]: See Array.empty, in shim-array.js

Comment (marchant) [inline generic-collection.js]: Thanks Kris, missed that, the problem is that there are many places where it should be used, (empty minus/plus) but these don’t require shim-array.js. Requiring it seems a bit wasteful and would increase the load time of collections a bit

Comment (marchant) [inline generic-collection.js]: Moving Array.empty to generic-collections.js would fix that

Comment (kriskowal) [inline generic-collection.js]: I agree, but if you’ve got FRB, you’ve got shims. I’m trying to get rid of shims in the v2 branch, though. If you go that way, isn’t too wasteful to just have a frozen array per module or to have a module with the frozen array.

Comment (marchant) [inline generic-collection.js]: frozen array per module sounds good, opportunity to flag them for debug as well

**Pull request merged**